### PR TITLE
Add modification to spreadsheet view

### PR DESCRIPTION
**Pull request by JacquesBorsenberger**

### CODE REVIEW

File: src/components/network/network-table.js [hunk 1]
```diff
@@ -159,6 +328,7 @@ const NetworkTable = (props) => {
                     props.network.twoWindingsTransformers[index]
                 }
                 columns={[
+                    makeHeaderCell('TWO_WINDING_TRANSFORMER'),
```
Comment (FranckLecuyer) [inline src/components/network/network-table.js:331]: WINDINGS (to be consistent with changes requested in https://github.com/gridsuite/network-modification-server/pull/10)

File: src/components/network/network-table.js [hunk 2]
```diff
@@ -238,6 +408,7 @@ const NetworkTable = (props) => {
                     props.network.threeWindingsTransformers[index]
                 }
                 columns={[
+                    makeHeaderCell('THREE_WINDING_TRANSFORMER'),
```
Comment (FranckLecuyer) [inline src/components/network/network-table.js:411]: WINDINGS (to be consistent with changes requested in gridsuite/network-modification-server#10)

File: src/utils/rest-api.js [hunk 3]
```diff
@@ -389,3 +399,30 @@ export function fetchAppsAndUrls() {
...
+    console.info('request network change');
+    const changeUrl =
+        getStudyUrl(studyName, userId) +
+        '/network-modification/' +
+        equipmentType +
+        '/' +
+        equipmentId;
+    console.debug(changeUrl);
+    return backendFetch(changeUrl, {
+        method: 'POST',
```
Comment (FranckLecuyer) [inline src/utils/rest-api.js:419]: POST or PUT ? (see remarks in gridsuite/network-modification-server#10)

File: src/components/study-pane.js [hunk 4]
```diff
@@ -430,26 +430,72 @@ const StudyPane = (props) => {
...
+            if (!typeEquipment || !equipmentId) {
+                return;
+            }
+            let fetchCb, equipments;
+            switch (typeEquipment) {
+                case 'GENERATOR':
+                    fetchCb = fetchGenerators(studyName, userId);
+                    equipments = network.generators;
+                    break;
+                case 'TWO_WINDING_TRANSFORMER':
```
Comment (FranckLecuyer) [inline src/components/study-pane.js:444]: WINDINGS (to be consistent with changes requested in gridsuite/network-modification-server#10)

File: src/components/study-pane.js [hunk 5]
```diff
@@ -430,26 +430,72 @@ const StudyPane = (props) => {
...
+                    fetchCb = fetchTwoWindingsTransformers(studyName, userId);
+                    equipments = network.twoWindingsTransformers;
+                    break;
+                case 'THREE_WINDING_TRANSFORMER':
```
Comment (FranckLecuyer) [inline src/components/study-pane.js:448]: WINDINGS (to be consistent with changes requested in gridsuite/network-modification-server#10)

File: src/components/study-pane.js [hunk 6]
```diff
@@ -430,26 +430,72 @@ const StudyPane = (props) => {
...
+                case 'loadflow_status':
+                    updateLFStatus();
+                    break;
+                case 'equipment':
+                    console.info(studyUpdatedForce.eventData.headers);
+                    updateEquipment(
+                        studyUpdatedForce.eventData.headers['equipment_type'] ||
+                            'GENERATOR',
+                        studyUpdatedForce.eventData.headers['equipment_id'] ||
+                            'PALISIN2'
```
Comment (FranckLecuyer) [inline src/components/study-pane.js:484]: PALISIN2 ????

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/utils/rest-api.js (hunk 3 above)
Comment (JacquesBorsenberger) [reply src/utils/rest-api.js:419]: post, it has a body

File: src/components/study-pane.js (hunk 6 above)
Comment (JacquesBorsenberger) [reply src/components/study-pane.js]: removed (value used for debug)

**Review by geofjamg — CHANGES_REQUESTED** [review]
I tried to modify a tap and a generator target and I had the following error in both cases:
```bash
TypeError: cr.command.replaceAll is not a function
commitChanges/<
src/components/network/network-table.js:103

  100 |     lineEdit[tabIndex].id.replace("'", "\\'") +
  101 |     "')\n";
  102 | Object.values(lineEdit[tabIndex].newValues).forEach((cr) => {
> 103 |     groovyCr += cr.command.replaceAll('{}', cr.value) + '\n';
      | ^  104 | });
  105 | requestNetworkChange(props.userId, props.studyName, groovyCr).then(
  106 |     (response) => {

commitChanges
src/components/network/network-table.js:102

   99 |     "('" +
  100 |     lineEdit[tabIndex].id.replace("'", "\\'") +
  101 |     "')\n";
> 102 | Object.values(lineEdit[tabIndex].newValues).forEach((cr) => {
      | ^  103 |     groovyCr += cr.command.replaceAll('{}', cr.value) + '\n';
  104 | });
  105 | requestNetworkChange(props.userId, props.studyName, groovyCr).then(

onClick
src/components/network/network-table.js:158

  155 | <Grid item>
  156 |     <IconButton
  157 |         size={'small'}
> 158 |         onClick={() => commitChanges(cellData.rowData)}
      | ^  159 |     >
  160 |         <CheckIcon />
  161 |     </IconButton>
```

**Review by geofjamg — APPROVED** [review]

**Pull request merged**